Sources/MapboxMaps/Foundation/Camera/BasicCameraAnimator.swift

```diff
@@ -122,20 +134,30 @@ public class BasicCameraAnimator: NSObject, CameraAnimator, CameraAnimatorInterf
         }
     }
 
-    /// Starts the animation after a delay. Once this method is called, there's no way to cancel the animation until it starts.
+    /// Starts the animation after a delay. This cannot be called on a paused animation.
+    /// If animations are cancelled before this animation has begun, it will also be cancelled.
     /// - Parameter delay: Delay (in seconds) after which the animation should start
     public func startAnimation(afterDelay delay: TimeInterval) {
-        delayedAnimationTimer = Timer.scheduledTimer(withTimeInterval: delay, repeats: false) { _ in
-            self.startAnimation()
+        if internalState != .initial {
+            fatalError("startAnimation(afterDelay:) cannot be called on paused, completed, or currently running animations.")
+        }
+
+        internalState = .delayed
+        delayedAnimationTimer = timerProvider(delay, false) { [weak self] (_) in
+            if let self = self {
+                self.internalState = .running(self.makeTransition())
```

Comment:
I'd like to change this so that `self.makeTransition()` is invoked at the beginning of the delay interval rather than at at the end of it. This is how `UIViewPropertyAnimator` seems to work.